### DISCUSSION TITLE
Fetch Row

### DISCUSSION THREAD

**Discussion by diselmxx** [Budibase/budibase#8815] [Help]
Hello, can you help me with next problem?

**Describe the bug**
I have async function getRow 
`const getRow = async () => {
    return await API.fetchRow({ tableId: users.tableId , rowId: userRowId })
  }`
When I call it in mount I get such an error "TypeError: e.enrichRows is not a function".
tableId and rowId are correct, I checked

**Comment by melohagan** [comment]
Hi @diselmxx 

Is this for a custom component?

**Reply by diselmxx** [reply]
@melohagan  Yes, it's my own component. I'm trying to create plugin

**Comment by melohagan** [comment]
Hey @diselmxx 

I'm not having any issues with the following:

```js
<script>
let val
  onMount(async () => { 
    val = await API.fetchRow({ tableId: "ta_123..." , rowId: "ro_ta_434..." }) 
  })
</script>

<div use:styleable={$component.styles}>
  Country: {val?.Country}
</div>
```

![Screenshot 2022-11-28 at 10 57 22](https://user-images.githubusercontent.com/101575380/204260965-4b896619-4a45-4800-a6c0-362c23acefd9.png)
